compiler/rustc_trait_selection/src/error_reporting/traits/on_unimplemented_condition.rs

```diff
@@ -1,52 +1,250 @@
-use rustc_ast::MetaItemInner;
-use rustc_attr_parsing as attr;
-use rustc_middle::ty::{self, TyCtxt};
+use rustc_ast::{MetaItemInner, MetaItemKind, MetaItemLit};
 use rustc_parse_format::{ParseMode, Parser, Piece, Position};
-use rustc_span::{DesugaringKind, Span, Symbol, kw, sym};
+use rustc_span::{DesugaringKind, Ident, Span, Symbol, kw, sym};
 
-/// A predicate in an attribute using on, all, any,
-/// similar to a cfg predicate.
+use crate::errors::InvalidOnClause;
+
+/// Represents the `on` filter in `#[rustc_on_unimplemented]`
 #[derive(Debug)]
 pub struct Condition {
-    pub inner: MetaItemInner,
+    span: Span,
```

Comment:
Could this be `pub(crate)` or `pub(super)` instead of `pub`? Likewise for a few other things below.

Reply:
Also, `Condition` is a very generic name. Could it be made more specific, e.g. `OnUnimplementedCondition`?